### PR TITLE
Use a global provider for optimistic purchases

### PR DESCRIPTION
**Pull request by valentinludu**
### WHAT
After purchase, show the success view and send the user a message when the transaction succeeded or failed on the blockchain.
If the user refreshes the page, we get all pending transactions and show a success message when they are done.
Update server components with import 'server-only'
e2e included

### WHY
We needed a way to optimistically update a user purchase

### CODE REVIEW

File: .env.test.local.example [hunk 1]
```diff
@@ -19,6 +19,6 @@ CREDENTIAL_WALLET_KEY=d093b8a44723eac658b4888743f945ff2066f1caba8c460f565002f998
 # Make playwright show color output when CI=true - https://github.com/microsoft/playwright/issues/28260
 FORCE_COLOR=1
 
-REACT_APP_BUILDER_NFT_CONTRACT_ADDRESS=0x1d305a06cb9dbdc32e08c3d230889acb9fe8a4dd
+REACT_APP_BUILDER_NFT_CONTRACT_ADDRESS=0x2cba9c6e0c14da826b0ec689cabf02a6f6b9808e
```
Comment (mattcasey) [inline .env.test.local.example:22]: this is not necessary right? we're not calling a real contract?

**Review by mattcasey — APPROVED** [review]

File: apps/scoutgame/playwright.config.ts [hunk 2]
```diff
@@ -13,13 +13,13 @@ import { devices } from '@playwright/test';
 const config: PlaywrightTestConfig = {
   testDir: './__e2e__',
   /* Maximum time one test can run for. */
-  timeout: 10 * 1000,
+  timeout: 10 * 10000,
```
Comment (mattcasey) [inline apps/scoutgame/playwright.config.ts:16]: 100 seconds is too long

**Review by mattcasey — CHANGES_REQUESTED** [review]
please clean up the test config

File: apps/scoutgame/playwright.config.ts (hunk 2 above)
Comment (valentinludu) [reply apps/scoutgame/playwright.config.ts]: I reverted everything from this file

File: .env.test.local.example (hunk 1 above)
Comment (valentinludu) [reply .env.test.local.example]: I reverted!

**Pull request merged**